src/main/java/frc/robot/Plunger.java

```diff
@@ -2,29 +2,224 @@
 
 //imports
 import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.Timer;
 
 /**
  *  Controls plunger mechanism.
  */
 public class Plunger{
 
     // object declaration
-    Solenoid solenoid;
+    Solenoid upstreamSolenoid;
+    Solenoid downstreamSolenoid;
+    AnalogInput pressureSensor;
+    AnalogInput vacuumSensor;
+    plungerState state;
+    Timer timer;
+    Joystick joy;
     
+
+    //Solenoid channels
+    public static final int UPSTREAM_SOLENOID_CHANNEL = 0;
+    public static final int DOWNSTREAM_SOLENOID_CHANNEL = 0;
+    public static final int PRESSURE_SENSOR_CHANNEL = 0;
+    public static final int VACUUM_SENSOR_CHANNEL = 0;
+
+    //Solenoid sensor variables
+    public static final double VACUUM_SENSOR_GOOD_VAC = 20;
+    public static final double VACUUM_SENSOR_MIN_VAC = 15; 
+
+    //joystick controls
+    public static final int PLUNGER_SUCTION_BUTTON = 0;
+
     /**
      * Constructs a new plunger object.
      */
-    public Plunger(){
+    public Plunger(DriveBase drive){
 
         // object initialization
-        solenoid = new Solenoid(0);
+        joy = drive.joyR;
+        upstreamSolenoid = new Solenoid(UPSTREAM_SOLENOID_CHANNEL);
+        downstreamSolenoid = new Solenoid(DOWNSTREAM_SOLENOID_CHANNEL);
+        pressureSensor = new AnalogInput(PRESSURE_SENSOR_CHANNEL);
+        vacuumSensor = new AnalogInput(VACUUM_SENSOR_CHANNEL);
+        timer = new Timer();
+        timer.start();
+    }
+
+    /**
+     * convert pressure sensor volts to psi
+     */
+    public double getPressure(){
+        return ((pressureSensor.getVoltage()*50.0)-25.0);
+    }
+
+    /**
+     * convert vacuum sensor volts to psi
+     */
+    public double getVacuum(){
+        return ((vacuumSensor.getVoltage()*11.125)-20.0625);
     }
 
+    /**
+     * declares plunger states
+     */
+    public enum plungerState {
+
+        VACUUM_ON,
+        DROP_STATE,
+        CLOSED,
+        VACUUM_TO_HOLD,
+        HOLD,
+        HOLD_TO_VACUUM
+    }
     
     /**
-     * Flips solenoid to other setting (T/F).
+     * control pulse, pickup, and dropping game piece 
      */
-    public void flipSolenoid(){
-        solenoid.set(!solenoid.get());
+    public void runPlunger(){
```

Comment:
All of this should be COMMENTED. For now it's fine but in the future please don't do this to yourself and the other programmers you have to work with :).